mrp_operations_extension/models/mrp_routing.py

```diff
@@ -0,0 +1,136 @@
+        string='Possible work centers for this operation')
+    do_production = fields.Boolean(
+        string='Produce here',
+        help="If enabled, the production and movement to stock of the final "
+             "products will be done in this operation. There can be only one "
+             "operation per route with this check marked.")
+    previous_operations_finished = fields.Boolean(
+        string='Previous operations finished',
+        default=default_previous_operations_finished)
+    picking_type_id = fields.Many2one('stock.picking.type', 'Picking Type',
+                                      domain=[('code', '=', 'outgoing')])
+
+    @api.constrains('op_wc_lines')
+    def _check_default_op_wc_lines(self):
+        if not self.op_wc_lines:
+            return
+        num_default = len([x for x in self.op_wc_lines if x.default])
+        if num_default != 1:
+            raise Warning(
+                _('There must be one and only one line set as default.'))
+
+    @api.one
+    @api.onchange('operation')
+    def onchange_operation(self):
+        if self.operation:
+            self.name = self.operation.name
+            self.note = self.operation.description
+            self.picking_type_id = self.operation.picking_type_id
+            self.op_wc_lines = False
+            op_wc_lst = []
+            is_default = True
+            for operation_wc in self.operation.workcenters:
+                data = {
+                    'default': is_default,
+                    'workcenter': operation_wc.id,
+                    'capacity_per_cycle': operation_wc.capacity_per_cycle,
+                    'time_efficiency': operation_wc.time_efficiency,
+                    'time_cycle': operation_wc.time_cycle,
+                    'time_start': operation_wc.time_start,
+                    'time_stop': operation_wc.time_stop,
+                    'op_number': self.operation.op_number,
+                }
+                op_wc_lst.append(data)
+                is_default = False
+            self.op_wc_lines = op_wc_lst
+        self.operation = False
+
+    @api.one
+    @api.onchange('op_wc_lines')
+    def onchange_lines_default(self):
+        for line in self.op_wc_lines:
+            if line.default:
+                self.workcenter_id = line.workcenter
+                if line.custom_data:
+                    self.cycle_nbr = line.capacity_per_cycle
+                    self.hour_nbr = line.time_cycle
+                else:
+                    self.cycle_nbr = line.workcenter.capacity_per_cycle
+                    self.hour_nbr = line.workcenter.time_cycle
+                break
+
+
+class MrpOperationWorkcenter(models.Model):
+    _name = 'mrp.operation.workcenter'
+    _description = 'MRP Operation Workcenter'
+
+    custom_data = fields.Boolean(
+        string="Custom", default=False,
+        help="If you mark this check, this means that the work center in this "
```

Comment:
"If checked, the work center in this routing will have a different capacity data than the one defined in the work center itself."